Improve error handling

mrkisaolamb opened the pull request:
To follow best practices for error and exception handling in Go and to prevent regressions, err113 and gosec checks in golang-ci-linter are now enabled. Errors now use predefined messages for status updates.
Resolve:  https://issues.redhat.com/browse/OSPRH-9063


Depends-On: https://github.com/openstack-k8s-operators/openstack-operator/pull/1306

stuggi commented on controllers/placementapi_controller.go:
```diff
@@ -66,6 +66,11 @@ import (
 
+const (
+	//NovaNetworkAttachmentsReadyInitMessage
+	NovaNetworkAttachmentsReadyErrorMessage = "NetworkAttachments error occurred not all pods have interfaces with ips as configured in NetworkAttachments: %s"
```
possibly not `Nova` ? see comment bellow, could we just update the one in lib-common?

mrkisaolamb replied:
done

stuggi commented on controllers/placementapi_controller.go:
```diff
@@ -1156,13 +1161,12 @@ func (r *PlacementAPIReconciler) ensureDeployment(
-			condition.NetworkAttachmentsReadyErrorMessage,
-			err.Error()))
+			NovaNetworkAttachmentsReadyErrorMessage,
+			instance.Spec.NetworkAttachments))
```
could we just update https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/common/condition/conditions.go#L269 ? now that it is only used as the condition error message?

mrkisaolamb replied:
done

mrkisaolamb commented:
/retest

mrkisaolamb commented:
/retest

mrkisaolamb commented:
https://github.com/openstack-k8s-operators/openstack-operator/pull/1306  landed, so we should be unblocked on ci

mrkisaolamb commented:
/retest

mrkisaolamb committed:
Improve error handling
To follow best practices for error and exception handling in Go and to prevent regressions,
err113 and gosec checks in golang-ci-linter are now enabled.

stuggi reviewed (approved):
/lgtm
